net/uring/io_uring.c

```diff
@@ -160,3 +160,23 @@ static go_completion_result completion(struct io_uring *ring, int block) {
     io_uring_cqe_seen(ring, cqe);
     return res;
 }
+
+typedef struct {
+  int probingNotSupported;
+  int supported;
+} probeCheck;
+
+
+// has_capability takes the index of a capability, and returns whether the operation is
+// supported, and if probing itself is supported.
+static probeCheck has_capability(int i) {
+    probeCheck pc = {0,0};
+    struct io_uring_probe *probe = io_uring_get_probe();
```

Comment:
You can probably do this entirely in Go/cgo, by calling `C.io_uring_get_probe` and `C.io_uring_opcode_supported`. That'd avoid needing to define a C struct etc.

The borderline between "write a C helper function" and "call individual C functions as needed" is a bit unclear. But for this case--performance doesn't matter, just two calls, need to return multiple values--it seems to me to be pretty clear.